x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/InferenceActionResponseTests.java

```diff
@@ -35,7 +36,12 @@ protected Writeable.Reader<InferenceAction.Response> instanceReader() {
 
     @Override
     protected InferenceAction.Response createTestInstance() {
-        return new InferenceAction.Response(List.of(TextExpansionResultsTests.createRandomResults()));
+        var result = switch (randomIntBetween(0, 1)) {
```

Comment:
Are there other tests that we should have to get coverage over that if-block in `InferenceAction.Response`?

Reply:
The if-block can be tested explicitly if you use `AbstractBWCWireSerializationTestCase` as the test base class. 

Implement the method `mutateInstanceForVersion` and you can simulate mixed version transport comms and assert on the expected output.

`AbstractBWCWireSerializationTestCase` is in xpack core and should be accessible here